### PR TITLE
[DOC] Fix installation link in readme and improve documentation build

### PR DESCRIPTION
**Pull request by mscheltienne**
Following on the comments in #57 

Changes:

- Fix the link to the installation instructions in the readme
- Remove `sphinx.ext.napoleon` which can conflicts with `numpydoc` (they do the same job, but the latter does more)
- Add `sphinx_copybutton` and `sphinx_design`
- Enable `nitpick` which will warn on any missing references during build
- Enable `numpydoc` docstring validation
- Rename deprecated `master_doc` to `root_doc`

TODO:

- [x] Add [sphinxcontrib-bibtex](https://github.com/mcmtroffaes/sphinxcontrib-bibtex) for the link to `michel 2018`
- [x] readme.rst in /tutorials/preprocessing says Below is a gallery of examples. without providing any examples. 

Depends on #44 as the readme has been extensively modified.

### CODE REVIEW

**Comment by mscheltienne** [comment]
@vferat FYI, the `readthedocs` build does not fail despite the `WARNING` issued. It would be nice to make it stricter. I tried adding a gh doc-build (not even working for now) just for testing and to see the warnings I have locally. Could you take a look at the `readthedocs` config?

**Comment by mscheltienne** [comment]
@vferat Ok, done. Now `readthedocs` should fail on warnings. 32 remaining!

**Comment by mscheltienne** [comment]
@vferat can you edit the 'About' on the main GH-page and add the link to the documentation website?

Screenshot 2022-08-12 at 14 33 14

**Comment by mscheltienne** [comment]
@vferat To sum up what has been done here until now:

- Set the DOC build workflow to fail on warnings/errors
- Enable nitpicking which warns on missing references
- Fix all the failing references ;)
- Add the `style.css` theme that Danial McCloy added to `mne-python` which improves a lot dark-theme readability
- Clean-up the `links.inc`
- Add `sphinxcontrib-bibtex` to manage bibliography
- Remove napoleon as numpydoc was also setup
- Add sphinx-copybutton and sphinx-design
- Rework the install page with tabs from sphinx-design
- Remove overview.rst and move it to the only place where it was included.

---------------

And also if you have time to look into this one, it's driving me crazy. I can't figure out how to either fix it or suppress it:

```
WARNING: multiple files found for the document "auto_tutorials/clustering/plot_modK": ['auto_tutorials\\clustering\\plot_modK.ipynb', 'auto_tutorials\\clustering\\plot_modK.py', 'auto_tutorials\\clustering\\plot_modK.py.md5', 'auto_tutorials\\clustering\\plot_modK.rst']
Use 'C:\\Users\\Mathieu\\Documents\\git\\mscheltienne\\pycrostates\\docs\\source\\auto_tutorials\\clustering\\plot_modK.rst' for the build.
WARNING: multiple files found for the document "auto_tutorials/group_level_analysis/plot_group_level_with_individual_clusters": ['auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_clusters.ipynb', 'auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_clusters.py', 'auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_clusters.py.md5', 'auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_clusters.rst']
Use 'C:\\Users\\Mathieu\\Documents\\git\\mscheltienne\\pycrostates\\docs\\source\\auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_clusters.rst' for the build.
WARNING: multiple files found for the document "auto_tutorials/group_level_analysis/plot_group_level_with_individual_gfp_peaks": ['auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_gfp_peaks.ipynb', 'auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_gfp_peaks.py', 'auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_gfp_peaks.py.md5', 'auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_gfp_peaks.rst']
Use 'C:\\Users\\Mathieu\\Documents\\git\\mscheltienne\\pycrostates\\docs\\source\\auto_tutorials\\group_level_analysis\\plot_group_level_with_individual_gfp_peaks.rst' for the build.
WARNING: multiple files found for the document "auto_tutorials/metrics/plot_metrics": ['auto_tutorials\\metrics\\plot_metrics.ipynb', 'auto_tutorials\\metrics\\plot_metrics.py', 'auto_tutorials\\metrics\\plot_metrics.py.md5', 'auto_tutorials\\metrics\\plot_metrics.rst']
Use 'C:\\Users\\Mathieu\\Documents\\git\\mscheltienne\\pycrostates\\docs\\source\\auto_tutorials\\metrics\\plot_metrics.rst' for the build.
WARNING: multiple files found for the document "auto_tutorials/preprocessing/plot_extract_gfp_peaks": ['auto_tutorials\\preprocessing\\plot_extract_gfp_peaks.ipynb', 'auto_tutorials\\preprocessing\\plot_extract_gfp_peaks.py', 'auto_tutorials\\preprocessing\\plot_extract_gfp_peaks.py.md5', 'auto_tutorials\\preprocessing\\plot_extract_gfp_peaks.rst']
Use 'C:\\Users\\Mathieu\\Documents\\git\\mscheltienne\\pycrostates\\docs\\source\\auto_tutorials\\preprocessing\\plot_extract_gfp_peaks.rst' for the build.
WARNING: multiple files found for the document "auto_tutorials/preprocessing/plot_resampling": ['auto_tutorials\\preprocessing\\plot_resampling.ipynb', 'auto_tutorials\\preprocessing\\plot_resampling.py', 'auto_tutorials\\preprocessing\\plot_resampling.py.md5', 'auto_tutorials\\preprocessing\\plot_resampling.rst']
Use 'C:\\Users\\Mathieu\\Documents\\git\\mscheltienne\\pycrostates\\docs\\source\\auto_tutorials\\preprocessing\\plot_resampling.rst' for the build.
```

Basically, it complains that it found several sources file, which is completely normal since they are generated by `sphinx-gallery`.

**Comment by mscheltienne** [comment]
doc-build should be green, but some links generated by linkcode should be broken, working on it..

**Comment by mscheltienne** [comment]
Ok nevermind, I'm giving up on `sphinx.ext.linkcode` for now, reverting to `sphinx.ext.viewcode`.
I could not figure out a good way to generate the URLs. I have it setup on 2 other repository fine.. but a couple of points are messing up the code I had for URL generation:
- Some methods are inherited from `mne`, so we have to handle both links pointing to MNE github and to Pycrostates github
- The verbose decorator, for some reason the fill_doc or copy_doc are fined, but the verbose decorator messes up the inspection to retrieve the fname. The code snippet below helps.. but it looks like it breaks the the detection of the correct lines for other methods/functions..

```
while hasattr(pyobject, '__wrapped__'):
            pyobject = pyobject.__wrapped__
```

- The project and the package have the same name, which leads to a duplicate in the file path: e.g. `"git/pycrostates/pycrostates/cluster/kmeans.py"`
- In combination with the previous points, the paths on `readthedocs` are a pain to work with as well. 

And I did not want to do some regex on a Saturday ;)

Anyway, @vferat please have a look at the render. Also please try to render it locally and compare, I think it renders differently on `readthedocs` that it does locally. Let me know what you think and what should be changed. I've spotted a couple more points to fix:

- [ ] The thumbnails of the gallery got lost
- [ ] On the install page, the numpy requirement should be 1.21

**Comment by vferat** [comment]
Great work! I tested locally and I have no error or warning for the build of the documentations. Both readthedoc and local version looks great.

I also fixed some references that did not exist or did not lead to the right URL.

I propose to merge the PR as it is (Fix docs building and rework website style) +
- Open a new PR for **Github links** as it may take some time to find a good solution especially if we need to redirect to both `MNE `and `pycrostates` repositories ( + the decorators issue).

- Open a new PR to review and improve the **content** of the documentation in response to the reviewers.

File: pycrostates/segmentation/segmentation.py [hunk 1]
```diff
@@ -315,9 +315,11 @@ def labels(self) -> NDArray[int]:
         """
         return self._labels.copy()
 
+    @fill_doc
```
Comment (mscheltienne) [inline pycrostates/segmentation/segmentation.py:318]: Why did you add this decorator here?

**Comment by mscheltienne** [comment]
Yes, works for me. I'll open an issue with what I have found so far for `linkcode`, at least we will keep it in mind.
Just check this decorator you added, and besides that good to merge!

**Comment by mscheltienne** [comment]
@vferat Looks like it did not run the doc-build and deploy on main after the merge?

**Comment by vferat** [comment]
The doc build failed because of this warning
`WARNING: conf value "version" should not be empty for EPUB3`

**Comment by mscheltienne** [comment]
Why is it trying to build `EPUB3` instead of ` html`? ...
I'll add `version = pycrostates.__version__` just in case, but it was not present before..

**Comment by vferat** [comment]
I disabled EPUB build, but the error is still here

**Comment by mscheltienne** [comment]
Ok I know, look at this file: https://github.com/vferat/pycrostates/blob/1c989877531e4284864f8b6ef8b969657fe8c897/.readthedocs.yml#L12-L13

**Comment by mscheltienne** [comment]
Should be fixed in this commit: 116ed1bdd15d9b1348c2449e49ecdd741be66615

**Comment by mscheltienne** [comment]
It's still not deploying..

**Comment by mscheltienne** [comment]
Nevermind, it did deploy, looks good!

**Comment by vferat** [comment]
Looks like it is working, however read the docs isn

_____________________
Victor Férat, PhD Student
________________________________

**Comment by mscheltienne** [comment]
?
